packages/application-generic/src/commands/project.command.ts

```diff
@@ -0,0 +1,32 @@
+import { IsNotEmpty } from 'class-validator';
+import { BaseCommand } from './base.command';
+
+export abstract class EnvironmentWithUserCommand extends BaseCommand {
+  @IsNotEmpty()
+  readonly environmentId: string;
+
+  @IsNotEmpty()
+  readonly organizationId: string;
+
+  @IsNotEmpty()
+  readonly userId: string;
+}
+
+export abstract class EnvironmentWithSubscriber extends BaseCommand {
+  @IsNotEmpty()
+  readonly environmentId: string;
+
+  @IsNotEmpty()
+  readonly organizationId: string;
+
+  @IsNotEmpty()
+  readonly subscriberId: string;
+}
```

Comment:
What do you think of extending both from `EnvironmentCommand` or do we need this extra flexibility?

Reply:
So this is just code I have moved from api to application generic and would not like to do fixes in it right now for this PR, because it is already massive. This is also used all over the project so would require a larger refactor....

Reply:
And I think we need the extra flexibility 😄